File: leetcode/24-swap-nodes-in-pairs/main.go
/*
Problem: Swap Nodes in Pairs
Link:	https://leetcode.com/problems/swap-nodes-in-pairs/description/
*/
package main


type ListNode struct {
  Val int
  Next *ListNode
}

func swapPairs(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
    return head
  }

  result := head.Next
  head.Next = swapPairs(head.Next.Next)
  result.Next = head

  return result
}

File: leetcode/24-swap-nodes-in-pairs/Main.kt
package main

class Solution {
    fun swapPairs(head: ListNode?): ListNode? {
        if (head?.next == null) {
            return head
        }

        val first = head
        val second = head.next
        val next = second?.next
        second?.next = first
        first?.next = swapPairs(next)

        return second
    }
}
